devito/data/data.py

```diff
@@ -132,16 +142,82 @@ def _global(self, glb_idx, decomposition):
         ret._is_distributed = any(i is not None for i in decomposition)
         return ret
 
+    def _check_idx(func):
+        """Check if __getitem__/__setitem__ may require communication across MPI ranks."""
+        @wraps(func)
+        def wrapper(data, *args, **kwargs):
+            glb_idx = args[0]
+            if len(args) > 1 and isinstance(args[1], Data) \
+                    and args[1]._is_mpi_distributed:
+                slicing = PARALLEL
+            elif data._is_mpi_distributed:
+                for i in as_tuple(glb_idx):
+                    if isinstance(i, slice) and i.step is not None and i.step < 0:
+                        slicing = PARALLEL
+                        break
+                    else:
+                        slicing = SERIAL
+            else:
+                slicing = SERIAL
+            kwargs['slicing'] = slicing
```

Comment:
slicing -> 'needs_comm' ?

Reply:
here and everywhere below in getitem/setitem

Reply:
`comm_on`? :p

Reply:
not sure if it's an english joke or ...? but this really got my attention :joy: